lib/elixir/lib/task.ex

```diff
@@ -341,10 +339,12 @@ defmodule Task do
   Receives a group of tasks and a message and finds
   a task that matches the given message.
 
-  This function returns a tuple with the returned value
-  in case the message matches a task that exited with
-  success alongside the matching task. It returns `nil`
-  if no task was found. It exits if the task has failed.
+  If the message matches a task that finished normally,
+  this function returns a tuple with the task's
+  result and `Task` struct. If the message matches a
+  task that terminated abnormally, this function will exit.
```

Comment:
OK, so I had to read the source code to finally understand what this function is doing. What do you think about the following wording:

```
  If the message matches a task that finished normally,
  this function returns a tuple with the task's
  result and `Task` struct. If the message matches a
  task that terminated before sending its result back,
  this function will exit.
```

I didn't want to say just "terminated" because that is more likely to raise question about the exact semantics.

As a slight offtopic, the exiting behaviour of this function surprises me. If the common use case for `find` is to check whether an incoming message is related to a task, I would expect to get an error tuple back for a prematurely terminated task. I guess this current behaviour is supposed to mirror what happens when you `await`?

/cc @josevalim